sections/advanced/theming.mdx

```diff
@@ -0,0 +1,117 @@
+import CodeBlock from '../../components/CodeBlock';
+import LiveEdit from '../../components/LiveEdit';
+
+## Theming
+
+styled-components has full theming support by exporting a `<ThemeProvider>` wrapper component.
+This component provides a theme to all React components underneath itself via the context API. In the render
+tree all styled-components will have access to the provided theme, even when they are multiple levels deep.
+
+To illustrate this, let's create our Button component, but this time we'll pass some variables down
+as a theme.
+
+<LiveEdit
+  code={'// Define our button, but with the use of props.theme this time\nconst Button = styled.button\`\n  font-size: 1em;\n  margin: 1em;\n  padding: 0.25em 1em;\n  border-radius: 3px;\n\n  /* Color the border and text with theme.main */\n  color: ${props => props.theme.main};\n  border: 2px solid ${props => props.theme.main};\n\`;\n\n// We are passing a default theme for Buttons that arent wrapped in the ThemeProvider\nButton.defaultProps = {\n  theme: {\n    main: "palevioletred"\n  }\n}\n\n// Define what props.theme will look like\nconst theme = {\n  main: "mediumseagreen"\n};\n\nrender(\n  <div>\n    <Button>Normal</Button>\n\n    <ThemeProvider theme={theme}>\n      <Button>Themed</Button>\n    </ThemeProvider>\n  </div>\n);\n  '}  noInline
+/>
```

Comment:
@timneutkens @jxnblk this is pretty annoying.

Can I somehow render a custom component for fenced codeblocks? That would allow us to just put a fenced code block here (like we had before) but still render it with out custom LiveEdit component if tagged with language `react`. (like we're doing right now)

Reply:
I believe that's called `code` and inline code is `inlineCode` when providing components

Reply:
I've also been disabling the `pre` rendering (like [here](https://github.com/jxnblk/mdx-go/blob/master/src/scope.js#L34-L35)) since it wraps `code` FWIW